dist-tools/service-loader.js

```diff
@@ -0,0 +1,169 @@
+var fs = require('fs');
+var util = require('util');
+var path = require('path');
+
+var AWS = require('../');
+var apis = require('../lib/api_loader');
+var metadata = require('../apis/metadata');
+
+var defaultServices = 'acm,apigateway,applicationautoscaling,autoscaling,cloudformation,cloudfront,cloudhsm,cloudtrail,cloudwatch,cloudwatchlogs,cloudwatchevents,codecommit,codedeploy,codepipeline,cognitoidentity,cognitoidentityserviceprovider,cognitosync,configservice,devicefarm,directconnect,dynamodb,dynamodbstreams,ec2,ecr,ecs,elasticache,elasticbeanstalk,elastictranscoder,elb,emr,firehose,gamelift,inspector,iotdata,kinesis,kms,lambda,marketplacecommerceanalytics,mobileanalytics,machinelearning,opsworks,rds,redshift,route53,route53domains,s3,ses,sns,sqs,ssm,storagegateway,sts,waf';
+var sanitizeRegex = /[^a-zA-Z0-9,-]/;
+
+var serviceClasses = {};
+Object.keys(AWS).forEach(function(name) {
+  if (AWS[name].serviceIdentifier) {
+    serviceClasses[AWS[name].serviceIdentifier] = AWS[name];
+  }
+});
+
+function getServiceHeader(service) {
+  if (service === 'all') {
+    return Object.keys(serviceClasses).map(function(name) {
+      return getServiceHeader(name);
+    }).join('\n');
+  }
+
+  if (!serviceClasses[service]) return null;
+  var versions = serviceClasses[service].apiVersions.map(function(version) {
+    return version.indexOf('*') >= 0 ? null : version;
+  }).filter(function(c) { return c !== null; });
+
+  var file = util.format(
+    'AWS.apiLoader.services[\'%s\'] = {};\n' +
+    'AWS.%s = AWS.Service.defineService(\'%s\', %s);\n',
+    service, apis.serviceName(service), service, util.inspect(versions));
+  var svcPath = path.join(__dirname, '..', 'lib', 'services', service + '.js');
+  if (fs.existsSync(svcPath)) {
+    file += 'require(\'./services/' + service + '\');\n';
+  }
+
+  return file;
+}
+
+function getService(service, version) {
+  if (service === 'all') {
+    return Object.keys(serviceClasses).map(function(name) {
+      var out = serviceClasses[name].apiVersions.map(function(svcVersion) {
+        if (svcVersion.indexOf('*') >= 0) return null;
+        return getService(name, svcVersion);
+      }).filter(function(c) { return c !== null; }).join('\n');
+
+      return out;
+    }).join('\n');
+  }
+
+  var svc, api;
+  if (!serviceClasses[service]) {
+    return null;
+  }
+
+  try {
+    var ClassName = serviceClasses[service];
+    svc = new ClassName({apiVersion: version, endpoint: 'localhost'});
+    api = apis.load(service, svc.api.apiVersion);
+  } catch (e) {
+    return null;
+  }
+
+  var serviceFileName = metadata[service].prefix || service;
+  var lines = [];
+  var line = util.format(
+    'AWS.apiLoader.services[\'%s\'][\'%s\'] = %s;\n',
```

Comment:
I'm working on rewriting this script to better support custom builds, so a lot will probably change here.

I'm not quite sure what you mean. Are you asking why we can't rely on browserify to pull in the service clients, or all the node.js module deps as well? Part of my motivation was so that webpack users wouldn't need browserify as well to customize their build.